moto/apigateway/utils.py

```diff
@@ -1,10 +1,81 @@
 import json
 import string
-from typing import Any, Dict
+from typing import Any, Dict, List, Union
 
 import yaml
 
 from moto.moto_api._internal import mock_random as random
+from moto.utilities.id_generator import ResourceIdentifier, Tags, generate_str_id
+
+
+class ApigwIdentifier(ResourceIdentifier):
+    service = "apigateway"
+
+    def __init__(self, account_id: str, region: str, name: str):
+        super().__init__(account_id, region, name)
+
+    def generate(
```

Comment:
It would be nice if we could move this method into the `ResourceIdentifier`-class with some sensible defaults, and only override it if necessary. 

It's currently difficult to tell what is sensible/common of course, without implementing many more `ResourceIdentifier`-classes, so that can wait until we have a better grasp of what identifiers typically look like.

Reply:
Good point, I will keep that in mind when expanding support to other resources.

On one side it would make using the `generate` method a lot simpler if it explicitly defines the property name(s) to pass.

But we will see, it may be that most will require `name` and will return the same `generate_str_id`.

ps. I have addressed the remaining comments in #8230